mathics/core/parser/parser.py

```diff
@@ -308,13 +308,33 @@ def p_RawLeftAssociation(self, token):
     def p_LeftRowBox(self, token):
         self.consume()
         children = []
-        self.box_depth += 1
-        self.bracket_depth += 1
+        # If this does not happend, it would be because
```

Comment:
happend -> happen

Reply:
I try to explain the problem in the comments, but it seems it is not easy. Here we go again: the problem is that
```\( a b \` c + d \)```
should be parsed as 
```FormBox[RowBox[{"c", "+", "d"}, RowBox[{"a", "b"}]]```

Then, to build the expression, we need to know all the tokens before ```\` ```, to build the "Format" expression, and then collect the following elements in the expression as the first argument. With the current parsing algorithm, I cannot make to `b_FormBox` what was the previous tokens. 
The alternative I found without changing the algorithm was to hack just `p_RawLeftAssociation`. If a ``` \` ```  token is found, then call again `p_RawLeftAssociation` but with a different tag. In this case, `self.box_deep` and `self.bracket_deep` does not change their values, unless a nested `Raw(Left|Right)Association` pair is found.

Reply:
As far as I could see, the unique case with this special behavior was `FormBox`. If there were more tags with a similar behavior, where to parsing the next token we need to look at several previous ones and reinterpret them, 
 I would consider rework the parser.

Reply:
I tried to make this more explicit in an extended comment  in
[f6f3d1f](https://github.com/Mathics3/mathics-core/pull/545/commits/f6f3d1fc29ed338d92829377f77d6a521e099129)

Also, I changed the comparison as @rocky suggested.